lasagne/layers/conv.py

```diff
@@ -249,7 +249,7 @@ def get_output_for(self, input, input_shape=None, **kwargs):
                                       filter_shape=self.get_W_shape(),
                                       border_mode='full')
             shift = (self.filter_size[0] - 1) // 2
-            conved = conved[:, :, shift:input.shape[2] + shift]
+            conved = conved[:, :, shift:-shift]
```

Comment:
> It won't work for even kernel sizes, but I don't think we were trying to support 'same' for those anyway.

I think we do. For the cases we don't, it's explicitly mentioned in the code (but I'm not sure about the documentation, we should document that and give a warning if we can).

As an alternative fix, what about using `input_shape[2] or input.shape[2]` instead of `input.shape[2]`? That will use the non-symbolic shape if it's not `None` (or `0`, but I think we don't need to distinguish that case). Maybe this would work just as well? If it does, I guess `Conv2DLayer` would need the same?